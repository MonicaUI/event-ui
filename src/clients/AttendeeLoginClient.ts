export default class AttendeeLoginClient {
  baseUrl: $TSFixMe;
  environment: $TSFixMe;
  eventId: $TSFixMe;
  userType: $TSFixMe;
  constructor(baseUrl: $TSFixMe, eventId: $TSFixMe, environment: $TSFixMe, userType: $TSFixMe) {
    this.baseUrl = baseUrl;
    this.eventId = eventId;
    this.environment = environment;
    this.userType = userType;
  }

  async authorize(): Promise<$TSFixMe> {
    window.location.assign(
      `${this.baseUrl}attendeeLogin/${this.eventId}/authorize?environment=${this.environment}&userType=${this.userType}`
    );
  }
}
