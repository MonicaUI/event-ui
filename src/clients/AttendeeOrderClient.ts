import { RequestBuilder } from '../utils/wrappedFetchAndRetryIfServerTooBusy';
import { SESSION_HEADER_EVENT_ID } from '@cvent/event-ui-networking';
import { fetchWithSessionTimeout } from '../dialogs/SessionTimedOutDialog';
import { ServiceError } from '@cvent/event-ui-networking';

export default class AttendeeOrderClient {
  attendeeOrderBaseUrl;
  environment;
  request;
  constructor(baseURL?: $TSFixMe, eventId?: $TSFixMe, environment?: $TSFixMe, authorizationToken?: $TSFixMe) {
    this.attendeeOrderBaseUrl = baseURL + 'attendee-order/v1/orders';
    this.environment = environment;
    this.request = new RequestBuilder({ url: this.attendeeOrderBaseUrl }).withCookies();
    this.request = this.request
      .header('Localized-Validations', 'true')
      .header(SESSION_HEADER_EVENT_ID, eventId)
      .auth(authorizationToken);
  }

  async getAttendeeOrders(eventId: $TSFixMe, attendeeId: $TSFixMe): Promise<$TSFixMe> {
    const request = this.request
      .url(`${this.attendeeOrderBaseUrl}`)
      .query('eventId', eventId)
      .query('attendeeId', attendeeId)
      .query('environment', this.environment)
      .get()
      .build();
    const response = await fetchWithSessionTimeout(request);
    if (!response.ok) {
      const error = await ServiceError.create('Error getting attendee orders', response, request);
      throw error;
    }
    return await response.json();
  }
}
