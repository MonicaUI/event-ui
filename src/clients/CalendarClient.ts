import { RequestBuilder } from '@cvent/nucleus-networking';
import { fetchAndRetryIfServerBusy } from '../utils/wrappedFetchAndRetryIfServerTooBusy';
import { ServiceError } from '@cvent/event-ui-networking';
import { ATTENDEE_TYPE } from '../../../../pkgs/event-widgets/constants/Attendee';

/**
 * The calendar service client to download iCal and google calendar event and session links.
 */
export default class CalendarClient {
  baseUrl: string;
  environment: string;
  eventId: string;
  constructor(baseUrl: string, eventId: string, environment: string) {
    this.baseUrl = `${baseUrl}calendar/v1/`;
    this.eventId = eventId;
    this.environment = environment;
  }
  _getRequestBuilder(urlPath = ''): RequestBuilder {
    const builder = new RequestBuilder({ url: `${this.baseUrl}${urlPath}` }).withCookies();
    return this.environment ? builder.query('environment', this.environment) : builder;
  }

  /**
   * Gets the event calendar url for the given calendarType
   * @param calendarType the calendar type supported such as ical, google
   * @param inviteeId the invitee for which the calendar link is requested
   * @param encodedInviteeId encoded invitee for which the calendar link is requested
   * @param previewToken previewToken for which the calendar link is requested
   * @param testModeHash testModeHash for which the calendar link is requested
   * @param cultureCode culture of the event
   * @returns {Promise<*>}
   */
  async getEventCalendar(
    calendarType: string,
    inviteeId: string,
    encodedInviteeId: string,
    previewToken: string,
    testModeHash: string,
    cultureCode: string
  ): Promise<{
    calendarType: string;
    calendarUrl: string;
  }> {
    const request = this._getRequestBuilder(`calendars/${calendarType}/events/${this.eventId}`)
      .query('inviteeId', inviteeId)
      .query('i', encodedInviteeId)
      .query('previewToken', previewToken)
      .query('tm', testModeHash)
      .query('locale', cultureCode)
      .get()
      .build();

    const response = await fetchAndRetryIfServerBusy(request);
    if (!response.ok) {
      const error = await ServiceError.create('getEventCalendar failed', response, request);
      throw error;
    }
    return await response.json();
  }

  /**
   * Gets the session calendar url for the given calendarType
   * @param calendarType the calendar type supported such as ical, google
   * @param sessionId the session for which the calendar link is requested
   * @param inviteeId the invitee for which the calendar link is requested
   * @param encodedInviteeId encoded invitee for which the calendar link is requested
   * @param previewToken previewToken for which the calendar link is requested
   * @param testModeHash testModeHash for which the calendar link is requested
   * @param locale culture of the event
   * @returns {Promise<*>}
   */
  async getSessionCalendar(
    calendarType: string,
    sessionId: string,
    inviteeId: string,
    encodedInviteeId: string,
    previewToken: string,
    testModeHash: string,
    locale: string
  ): Promise<{
    calendarType: string;
    calendarUrl: string;
  }> {
    const request = this._getRequestBuilder(`calendars/${calendarType}/events/${this.eventId}/sessions/${sessionId}`)
      .query('inviteeId', inviteeId)
      .query('i', encodedInviteeId)
      .query('previewToken', previewToken)
      .query('tm', testModeHash)
      .query('locale', locale)
      .get()
      .build();

    const response = await fetchAndRetryIfServerBusy(request);
    if (!response.ok) {
      const error = await ServiceError.create('getSessionCalendar failed', response, request);
      throw error;
    }
    return await response.json();
  }

  /**
   * Redirects the user either to the VirtualDetails page for webcast details or VAH depending upon conditions.
   * @param sessionId sessionId to which redirection needs to be done.
   * @param attendeeId attendee requesting to redirect.
   * @param attendeeType type of attendee requesting to redirect.
   * @returns {Promise<*>}
   */
  async redirectToVirtualEvent(sessionId: string, attendeeId: string, attendeeType: string): Promise<void> {
    const attendeeQueryParam = this.isPrimaryInvitee(attendeeType) ? 'inviteeId' : 'g';
    const request =
      this.baseUrl +
      `calendars/events/${this.eventId}/virtualSessionLink?` +
      `${attendeeQueryParam}=${attendeeId}` +
      `&sessionId=${sessionId}`;
    window.open(request, '_blank');
  }

  isPrimaryInvitee(attendeeType: string): boolean {
    return attendeeType === ATTENDEE_TYPE.ATTENDEE || attendeeType === ATTENDEE_TYPE.GROUP_LEADER;
  }
}
