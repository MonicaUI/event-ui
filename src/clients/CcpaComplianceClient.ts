import { RequestBuilder, fetchAndRetryIfServerBusy } from '../utils/wrappedFetchAndRetryIfServerTooBusy';
import { SESSION_HEADER_EVENT_ID } from '@cvent/event-ui-networking';

/**
 * The client to post CCPA compliance request.
 */
export default class CcpaComplianceClient {
  baseUrl: $TSFixMe;
  environment: $TSFixMe;
  eventId: $TSFixMe;
  constructor(baseUrl: $TSFixMe, eventId: $TSFixMe, environment: $TSFixMe) {
    this.baseUrl = baseUrl;
    this.eventId = eventId;
    this.environment = environment;
  }
  _getRequestBuilder(urlPath = ''): $TSFixMe {
    const builder = new RequestBuilder({ url: `${this.baseUrl}${urlPath}` })
      .header(SESSION_HEADER_EVENT_ID, this.eventId)
      .withCookies();
    return this.environment ? builder.query('environment', this.environment) : builder;
  }
  /**
   * Post CCPA compliance request.
   * @param eventId - eventId of the event for which ccpa opt out request is made
   * @param ccpaComplianceRequests List of requests for which opt out request needs to be made
   */
  async makeCcpaComplianceRequest(ccpaComplianceRequests: $TSFixMe): Promise<$TSFixMe> {
    const request = this._getRequestBuilder('compliance/ccpa')
      .query('eventId', this.eventId)
      .post()
      .json(ccpaComplianceRequests)
      .build();
    const response = await fetchAndRetryIfServerBusy(request);
    return { isSuccessful: response.status === 204 };
  }
}
