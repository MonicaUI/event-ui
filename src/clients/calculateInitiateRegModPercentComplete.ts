/**
 * Calculate progress for reg mod creation. Position starts at an arbitrary
 * number (initialPosition) and counts towards zero.
 */
export default function calculateInitiateRegModPercentComplete(
  initialPosition: $TSFixMe,
  position: $TSFixMe
): $TSFixMe {
  return 100 * (initialPosition === 0 ? 1 : 1 - position / initialPosition);
}
