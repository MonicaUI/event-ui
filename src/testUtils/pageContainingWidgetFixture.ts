export default function pageContainingWidgetFixture(pageId: $TSFixMe, widgetId: $TSFixMe): $TSFixMe {
  const pages = {
    [pageId]: {
      id: pageId,
      name: pageId,
      title: pageId,
      version: 1,
      rootLayoutItemIds: ['container'],
      type: 'PAGE'
    }
  };
  const layoutItems = {
    container: {
      id: 'container',
      layout: {
        type: 'container',
        parentId: null,
        childIds: [widgetId]
      }
    },
    [widgetId]: {
      id: widgetId,
      layout: {
        type: 'page',
        parentId: 'container',
        childIds: []
      },
      config: {}
    }
  };
  return { pages, layoutItems };
}
