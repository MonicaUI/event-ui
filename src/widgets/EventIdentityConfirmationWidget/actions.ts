import {
  getEventRegistration,
  getVisibleRegistrationTypesByPlannerSelection
} from '../../redux/selectors/currentRegistrant';
import { getRegistrationPageFields } from '../../redux/selectors/currentRegistrationPath';
import { NOT_VISIBLE, READ_ONLY } from 'cvent-question-widgets/lib/DisplayType';
import { isWidgetReviewed } from '../../redux/website/pageContentsWithGraphQL';
import StandardContactFields from 'event-widgets/lib/StandardContactFields/StandardContactFields';
import { defaultRegistrationTypeId } from 'event-widgets/utils/registrationType';
import { CLEAR_REG_CART_INFERRED_FIELDS } from '../../redux/registrationForm/regCart/actionTypes';
import { setIn } from 'icepick';
import { getRegCart } from '../../redux/selectors/shared';
import { saveRegistration } from '../../redux/registrationForm/regCart';
import { runRegistrationTypeChangeValidationsForPrimaryAndGuest } from '../../redux/registrationForm/regCart';
import {
  handleRegistrationTypeSelectionConflict,
  validateUserRegistrationTypeSelection
} from '../../dialogs/selectionConflictDialogs';
import { withLoading } from 'nucleus-guestside-site/src/redux/modules/dialogContainer';
import { getUpdateErrors } from '../../redux/registrationForm/errors';
import { openNoAdmissionItemAvailableForRegistrationTypeDialog, openPrivateEventErrorDialog } from '../../dialogs';
import { merge } from 'lodash';
import { populateVisibleProducts } from '../../redux/visibleProducts';
import { clearOutOfSyncUserSessionFieldsAndUrlQuery } from '../../redux/registrantLogin/actions';
import { ApolloClient } from '@apollo/client';
import type { RootState, AsyncAppThunk } from '../../redux/reducer';

export const isRegTypeReviewed = async (
  state: RootState,
  registrationTypeId: string,
  registrationPathId: string,
  apolloClient: ApolloClient<unknown>
): Promise<boolean> => {
  const widgetTypeName = 'RegistrationType';
  const visibleRegTypes = getVisibleRegistrationTypesByPlannerSelection(state, registrationPathId) || [];
  const isRegTypeWidgetReviewed = await isWidgetReviewed(state, { widgetType: widgetTypeName }, apolloClient);
  return visibleRegTypes.includes(registrationTypeId) && isRegTypeWidgetReviewed;
};

export function clearFields(): AsyncAppThunk {
  return async (
    dispatch,
    getState,
    {
      apolloClient
    }: {
      apolloClient?: ApolloClient<unknown>;
    }
  ) => {
    try {
      const state = getState();
      const eventRegistration = JSON.parse(JSON.stringify(getEventRegistration(state)));
      const isRegTypeWidgetReviewed = await isRegTypeReviewed(
        state,
        eventRegistration.registrationTypeId,
        eventRegistration.registrationPathId,
        apolloClient
      );
      if (state.userSession.regTypeId && !isRegTypeWidgetReviewed) {
        const regTypeChangeValidations = await runRegistrationTypeChangeValidationsForPrimaryAndGuest(
          validateUserRegistrationTypeSelection,
          defaultRegistrationTypeId,
          getState(),
          true,
          apolloClient
        );
        // eslint-disable-next-line @typescript-eslint/prefer-optional-chain
        if (regTypeChangeValidations && regTypeChangeValidations.hasConflicts) {
          return dispatch(
            handleRegistrationTypeSelectionConflict(
              regTypeChangeValidations.validationResults,
              null,
              null,
              dissociateInviteeAndClearInferredFields
            )
          );
        }
      }
      const linkedInviteeIdToRemove = eventRegistration.attendee.attendeeId;
      if (state.userSession.regTypeId && !isRegTypeWidgetReviewed) {
        eventRegistration.registrationTypeId = defaultRegistrationTypeId;
      }
      /*
       * This is an existing pattern that we will need to revisit later. We should clear
       * the individual field rather than replace the entire reg cart
       */
      dispatch({
        type: CLEAR_REG_CART_INFERRED_FIELDS,
        payload: {
          regCart: setIn(
            getRegCart(state),
            ['eventRegistrations', eventRegistration.eventRegistrationId],
            eventRegistration
          )
        }
      });
      /**
       * on clear fields regType is cleared, need to reevaluate visible products
       */
      await dispatch(populateVisibleProducts(eventRegistration.eventRegistrationId));
      await dispatch(dissociateInviteeAndClearInferredFields());
      return await dispatch(withLoading(saveRegistration)(linkedInviteeIdToRemove));
    } catch (error) {
      if (
        // @ts-expect-error ts-migrate(2554) FIXME: Expected 1 arguments, but got 2.
        getUpdateErrors.isPrivateEvent(error, getState()) ||
        // @ts-expect-error ts-migrate(2554) FIXME: Expected 0-1 arguments, but got 2.
        getUpdateErrors.isAttendeeNotAllowedByCustomLogic(error, getState())
      ) {
        return await dispatch(openPrivateEventErrorDialog());
      }
      if (getUpdateErrors.isAdmissionItemsNotAvailableForRegTypeError(error)) {
        return await dispatch(openNoAdmissionItemAvailableForRegistrationTypeDialog());
      }
      throw error;
    }
  };
}

export function dissociateInviteeAndClearInferredFields(): AsyncAppThunk {
  return async (dispatch, getState) => {
    const state = getState();
    const eventRegistration = JSON.parse(JSON.stringify(getEventRegistration(state)));
    // invitee associated attendeeId and contact Id
    delete eventRegistration.attendee.attendeeId;
    delete eventRegistration.attendee.personalInformation.contactId;
    // inferred information to clear: 2. inviteeId related personalInformation
    const existingContactInfo = eventRegistration.attendee.personalInformation;
    const contactInfo = await Promise.all(
      getRegistrationPageFields(state).map(async field => {
        const isFieldReviewed = await isWidgetReviewed(state, { fieldId: field.fieldId });
        return field.display !== NOT_VISIBLE && field.display !== READ_ONLY && isFieldReviewed ? field : null;
      })
    );
    const filteredContactInfo = contactInfo.filter(Boolean);
    const copiedContactInfo = filteredContactInfo.map(field =>
      field.isCustomField
        ? {
            customFields: {
              [field.fieldId]: existingContactInfo.customFields[field.fieldId]
            }
          }
        : {
            // @ts-expect-error ts-migrate(2464) FIXME: A computed property name must be of type 'string',... Remove this comment to see the full error message
            [(StandardContactFields[field.fieldId] || {}).regApiPath]:
              // @ts-expect-error ts-migrate(2538) FIXME: Type 'string[]' cannot be used as an index type.
              existingContactInfo[(StandardContactFields[field.fieldId] || {}).regApiPath]
          }
    );
    const temp = copiedContactInfo.reduce(
      (copiedInfo, info) => (info.customFields ? merge(copiedInfo, info) : Object.assign(copiedInfo, info)),
      {}
    );
    eventRegistration.attendee.personalInformation = temp;

    await dispatch(clearOutOfSyncUserSessionFieldsAndUrlQuery());
    // Reversing the order so that inviteeId is set to undefined for sourceId to be shown as editable.
    dispatch({
      type: CLEAR_REG_CART_INFERRED_FIELDS,
      payload: {
        regCart: setIn(
          getRegCart(state),
          ['eventRegistrations', eventRegistration.eventRegistrationId],
          eventRegistration
        )
      }
    });
  };
}
