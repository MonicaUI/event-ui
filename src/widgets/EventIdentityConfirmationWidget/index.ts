import { connect } from 'react-redux';
import EventIdentityConfirmationWidget from 'event-widgets/lib/IdentityConfirmation/IdentityConfirmationWidget';
import { updateAdmin } from '../../redux/registrationForm/regCart/actions';
import { isLinkedInviteeRegistration } from '../../redux/userSession';
import { isPlannerRegistration } from '../../redux/defaultUserSession';
import { CLOSED } from 'event-widgets/clients/EventStatus';
import {
  getEventRegistrationId,
  isRegistrationModification,
  isGroupMember,
  getAdminPersonalInformation,
  isGroupRegistration,
  isOnPrivateRegPathOrWithLimitedTargetList,
  isOnEmailInviteOnlyRegPath
} from '../../redux/selectors/currentRegistrant';
import {
  isAdministratorRegistrationEnabled,
  shouldMakeAdminFieldReadOnly,
  isSsoInviteeFlow,
  isHTTPPostOrSSOOnInAccount,
  isOAuthOnInAccount,
  areRegistrationActionsDisabled
} from '../../redux/selectors/shared';
import { getRegistrationPathIdForWidget } from 'event-widgets/redux/selectors/website/pluginData/registrationProcessNavigation';
import { clearFields } from './actions';
import { isExternalAuthOnInEvent, isOAuthOnInEvent } from '../../redux/selectors/event';

const isOAuthOn = (account, event) => {
  return isOAuthOnInAccount(account) && isOAuthOnInEvent(event);
};

/**
 * Data wrapper for the event identity confirmation widget.
 */
export default connect(
  (state: $TSFixMe, props: $TSFixMe) => {
    const eventRegistrationId = getEventRegistrationId(state);
    const isRegMod = isRegistrationModification(state);
    const groupMember = isGroupMember(state, eventRegistrationId);
    const regPathId = getRegistrationPathIdForWidget(state, props.id);
    return {
      // @ts-expect-error ts-migrate(2339) FIXME: Property 'getConfig' does not exist on type 'typeo... Remove this comment to see the full error message
      config: EventIdentityConfirmationWidget.getConfig(state, props.config, 'EventIdentityConfirmation', props.id),
      shouldMakeAdminFieldReadOnly: shouldMakeAdminFieldReadOnly(state),
      showAdminRegCheckbox:
        isAdministratorRegistrationEnabled(state, regPathId) &&
        !isRegMod &&
        !groupMember &&
        !isSsoInviteeFlow(state, regPathId),
      isExternalAuthEnabled: isHTTPPostOrSSOOnInAccount(state.account) && isExternalAuthOnInEvent(state.event),
      isOAuthEnabled: isOAuthOn(state.account, state.event),
      admin: getAdminPersonalInformation(state) || { selectedValue: false },
      hideClearButton:
        isGroupRegistration(state) ||
        !isLinkedInviteeRegistration(state) ||
        isPlannerRegistration(state) ||
        isRegMod ||
        state.event.status === CLOSED ||
        (isOnPrivateRegPathOrWithLimitedTargetList(state) && isOnEmailInviteOnlyRegPath(state)) ||
        areRegistrationActionsDisabled(state),
      isSiteEditor: false
    };
  },
  {
    onClear: clearFields,
    onAdminUpdate: updateAdmin
  },
  (stateProps: $TSFixMe, dispatchProps: $TSFixMe, ownProps: $TSFixMe) => ({
    ...ownProps,
    ...stateProps,
    ...dispatchProps,
    onClear: stateProps.hideClearButton ? undefined : dispatchProps.onClear,
    onAdminUpdate: dispatchProps.onAdminUpdate,
    hideClearButton: undefined
  })
)(EventIdentityConfirmationWidget);
