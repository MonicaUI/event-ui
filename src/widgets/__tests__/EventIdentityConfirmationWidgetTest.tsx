import React from 'react';
import EventIdentityConfirmationWidget from '../EventIdentityConfirmationWidget';
import { shallow } from 'enzyme';
import pageContainingWidgetFixture from '../../testUtils/pageContainingWidgetFixture';
import { setIn } from 'icepick';
import { PRIVATE_ALL_TARGETED_LISTS, PRIVATE_LIMITED_TARGETED_LISTS } from 'event-widgets/clients/RegistrationPathType';
import { ACTIVE } from '../../../../../pkgs/event-widgets/clients/EventStatus';

const loadMetaDataFunction = jest.fn(() => {
  return {};
});
let regMod = false;
let attendeeType = 'GROUP_LEADER';
function getState() {
  return {
    registrationForm: {
      regCart: {
        regMod,
        groupRegistration: false,
        eventRegistrations: {
          eventRegId: {
            registrationPathId: 'regPathId',
            eventRegistrationId: 'eventRegId',
            attendeeType
          }
        }
      }
    },
    website: {
      ...pageContainingWidgetFixture('pageId', 'widgetId'),
      pluginData: {
        registrationProcessNavigation: {
          registrationPaths: {
            regPathId: {
              id: 'regPathId',
              pageIds: ['pageId']
            }
          }
        }
      },
      siteInfo: {
        sharedConfigs: {}
      }
    },
    appData: {
      registrationSettings: {
        registrationPaths: {
          regPathId: {
            registrationTypeSettings: {
              limitVisibility: false
            },
            guestRegistrationSettings: {
              registrationTypeSettings: {
                limitVisibility: false
              }
            },
            identityConfirmation: {
              administratorRegistrationEnabled: true
            }
          }
        }
      }
    },
    widgetFactory: {
      loadMetaData: loadMetaDataFunction
    },
    userSession: {
      inviteeId: 'fake-invitee-id'
    },
    defaultUserSession: {},
    event: {
      status: ACTIVE
    },
    account: {
      settings: {}
    }
  };
}
const subscribe = () => {};
async function dispatch(action) {
  if (typeof action === 'function') {
    await action(dispatch, getState);
  }
}
const props = {
  store: { getState, subscribe, dispatch },
  translate: (res, opts) => (opts ? `${res}:${JSON.stringify(opts)}` : res),
  style: {},
  config: {
    displayText: ''
  },
  classes: {},
  id: 'widgetId'
};

describe('EventIdentityConfirmationWidget', () => {
  test('should render with admin fields', () => {
    const widget = shallow(<EventIdentityConfirmationWidget {...props} />);
    expect(widget).toMatchSnapshot();
  });

  test('should not render admin fields during reg mod', () => {
    regMod = true;
    const widget = shallow(<EventIdentityConfirmationWidget {...props} />);
    expect(widget).toMatchSnapshot();
    regMod = false;
  });

  test('should not render admin fields for group member', () => {
    attendeeType = 'ATTENDEE';
    const widget = shallow(<EventIdentityConfirmationWidget {...props} />);
    expect(widget).toMatchSnapshot();
    attendeeType = 'GROUP_LEADER';
  });
  test('should not render clear field if event is private and EmailOnly', () => {
    let initialState = getState();
    initialState = setIn(
      initialState,
      [
        'appData',
        'registrationSettings',
        'registrationPaths',
        'regPathId',
        'accessRules',
        'invitationListAccess',
        'type'
      ],
      PRIVATE_ALL_TARGETED_LISTS
    );
    initialState = setIn(
      initialState,
      [
        'appData',
        'registrationSettings',
        'registrationPaths',
        'regPathId',
        'accessRules',
        'invitationListAccess',
        'isEmailOnlyInvite'
      ],
      true
    );

    function getInitialState() {
      return initialState;
    }
    const defaultProps = {
      store: { getState: getInitialState, subscribe, dispatch },
      translate: (res, opts) => (opts ? `${res}:${JSON.stringify(opts)}` : res),
      style: {},
      config: {
        displayText: ''
      },
      classes: {},
      id: 'widgetId'
    };
    const wrapper = shallow(<EventIdentityConfirmationWidget {...defaultProps} />).dive();
    expect(wrapper.find('IdentityConfirmationWidget').props().onClear).toBe(undefined);
  });
  test('should not render clear field if the person is on a select invitation list(Private Event) and EmailOnly', () => {
    let initialState = getState();
    initialState = setIn(
      initialState,
      [
        'appData',
        'registrationSettings',
        'registrationPaths',
        'regPathId',
        'accessRules',
        'invitationListAccess',
        'type'
      ],
      PRIVATE_LIMITED_TARGETED_LISTS
    );
    initialState = setIn(
      initialState,
      [
        'appData',
        'registrationSettings',
        'registrationPaths',
        'regPathId',
        'accessRules',
        'invitationListAccess',
        'isEmailOnlyInvite'
      ],
      true
    );

    function getInitialState() {
      return initialState;
    }
    const defaultProps = {
      store: { getState: getInitialState, subscribe, dispatch },
      translate: (res, opts) => (opts ? `${res}:${JSON.stringify(opts)}` : res),
      style: {},
      config: {
        displayText: ''
      },
      classes: {},
      id: 'widgetId'
    };
    const wrapper = shallow(<EventIdentityConfirmationWidget {...defaultProps} />).dive();
    expect(wrapper.find('IdentityConfirmationWidget').props().onClear).toBe(undefined);
  });
  test('should disable the admin check box in SSO invitee flow', () => {
    const initialState = getState();
    const newState = {
      ...initialState,
      registrationForm: {
        ...initialState.registrationForm,
        regCart: {
          ...initialState.registrationForm.regCart,
          admin: {
            disableAdminRegistrationInSso: true
          }
        }
      },
      userSession: {
        ...initialState.userSession,
        isSsoAdmin: false
      },
      account: {
        settings: {
          accountSecuritySettings: {
            allowHTTPPost: false,
            allowSSOLogin: true,
            allowSecureHTTPPost: true
          }
        }
      },
      event: {
        eventSecuritySetupSnapshot: {
          authenticationType: 1
        }
      }
    };

    function getInitialState() {
      return newState;
    }

    const newDefaultProps = {
      store: { getState: getInitialState, subscribe, dispatch },
      translate: (res, opts) => (opts ? `${res}:${JSON.stringify(opts)}` : res),
      style: {},
      config: {
        displayText: ''
      },
      classes: {},
      id: 'widgetId'
    };
    const wrapper = shallow(<EventIdentityConfirmationWidget {...newDefaultProps} />).dive();
    expect(wrapper.find('IdentityConfirmationWidget').prop('showAdminRegCheckbox')).toBeFalsy();
  });
  test('should not render clear field if external Auth is enabled in event', () => {
    const initialState = getState();

    const newState = {
      ...initialState,
      account: {
        settings: {
          accountSecuritySettings: {
            allowHTTPPost: true,
            allowSSOLogin: false,
            allowSecureHTTPPost: false
          }
        }
      },
      event: {
        eventSecuritySetupSnapshot: {
          authenticationType: 1
        }
      }
    };

    function getInitialState() {
      return newState;
    }
    const defaultProps = {
      store: { getState: getInitialState, subscribe, dispatch },
      translate: (res, opts) => (opts ? `${res}:${JSON.stringify(opts)}` : res),
      style: {},
      config: {
        displayText: ''
      },
      classes: {},
      id: 'widgetId'
    };
    const wrapper = shallow(<EventIdentityConfirmationWidget {...defaultProps} />).dive();
    expect(wrapper.find('IdentityConfirmationWidget').props().isExternalAuthEnabled).toBeTruthy();
  });
});
