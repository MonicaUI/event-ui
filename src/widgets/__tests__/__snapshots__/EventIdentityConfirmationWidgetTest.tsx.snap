// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventIdentityConfirmationWidget should not render admin fields during reg mod 1`] = `ShallowWrapper {}`;

exports[`EventIdentityConfirmationWidget should not render admin fields for group member 1`] = `ShallowWrapper {}`;

exports[`EventIdentityConfirmationWidget should render with admin fields 1`] = `ShallowWrapper {}`;
